### PR TITLE
Be more positive with setting labels

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/6435

This is done through an on-the-fly inverter for the settings. All the settings changed are boolean values, so this should be more than safe to just let happen throughout the SettingsStore. Typically a change like this would be done in the individual handlers (similar to how setting names are remapped to different properties or even different storage locations on the fly), however doing that for this many settings would be a huge nightmare and involve changing *all* the layers. By putting a global "invert this" flag on the setting, we can get away with doing the inversion as the last possible step during a read (or write).

To speed up calculations of the default values, we cache all the inverted values into a lookup table similar to how we represent the defaults already. Without this, the DefaultHandler would need to iterate the setting list and invert the values, slowing things down over time. We invert the value up front so we can keep the generic inversion logic without checking the level ahead of time. It is fully intended that a default value represents the new setting name, not the legacy name.

This commit also includes a debugger for settings because it was hard to visualize what the SettingsStore was doing during development. Some added information is included as it may be helpful for when someone has a problem with their settings and we need to debug it. Typically the debugger would be run in conjunction with `mxSendRageshake`: `mxSettingsStore.debugSetting('showJoinLeaves') && mxSendRageshake('Debugging showJoinLeaves setting')`.

### CODE REVIEW

File: src/settings/SettingsStore.js [hunk 1]
```diff
@@ -220,34 +226,54 @@ export default class SettingsStore {
...
+        }
+
         if (explicit) {
             const handler = handlers[level];
-            if (!handler) return SettingsStore._tryControllerOverride(settingName, level, roomId, null, null);
-            const value = handler.getValue(settingName, roomId);
-            return SettingsStore._tryControllerOverride(settingName, level, roomId, value, level);
+            if (!handler) {
+                let value = SettingsStore._tryControllerOverride(setting, level, roomId, null, null);
+                if (inverted) value = !value;
```
Comment (bwindels) [inline src/settings/SettingsStore.js]: Hmm, slightly concerned about this important line being duplicated 4x in the code below...

**Comment by turt2live** [comment]
> migrating the inverted settings all in one go at startup

This has the problem of breaking people's ability to downgrade or switch between versions of riot. Most of the settings are cosmetic things, however some are privacy related. I'd be very uncomfortable with risking people not realizing what happened.

**Comment by bwindels** [comment]
> > migrating the inverted settings all in one go at startup
> 
> This has the problem of breaking people's ability to downgrade or switch between versions of riot. Most of the settings are cosmetic things, however some are privacy related. I'd be very uncomfortable with risking people not realizing what happened.

Good point

**Comment by turt2live** [comment]
@bwindels please take another look - I've renamed `_tryControllerOverride` to `_getFinalValue` which does the inversion there instead of everywhere.

**Comment by lampholder** [comment]
This PR gives me great joy :D

**Pull request merged**